File: frontend/src/types/movie.ts
export interface Movie {
	imdbID: string;
	Title: string;
	Year: string;
	Type: string;
	Poster: string;
	// Optional — present when full details have been fetched
	Genre?: string;
	imdbRating?: string;
	Runtime?: string;
	Director?: string;
	Actors?: string;
	Plot?: string;
	Language?: string;
	Country?: string;
	Awards?: string;
	Response?: string;
	Error?: string;
}

export interface MovieDetails {
	imdbID: string;
	Title: string;
	Year: string;
	Rated: string;
	Released: string;
	Runtime: string;
	Genre: string;
	Director: string;
	Writer: string;
	Actors: string;
	Plot: string;
	Language: string;
	Country: string;
	Awards: string;
	Poster: string;
	Ratings: Array<{
		Source: string;
		Value: string;
	}>;
	Metascore: string;
	imdbRating: string;
	imdbVotes: string;
	Type: string;
	DVD: string;
	BoxOffice: string;
	Production: string;
	Website: string;
	Response: string;
	Error?: string;
}

export interface SearchResponse {
	Search: Movie[];
	totalResults: string;
	Response: string;
	Error?: string;
}

export interface UserRating {
	movieId: string;
	rating: number;
}

// Actor-related types
export interface Actor {
	name: string;
	movies: string[];
	genres: string[];
	birthYear: number;
	nationality: string;
}

export interface ActorSearchResult {
	actor: Actor;
	matchedMovies: Movie[];
	totalMovies: number;
}

export interface SearchType {
	type: 'movie' | 'actor';
	query: string;
}

File: backend/movie-service/src/main/java/com/cineverse/movie/model/Movie.java
package com.cineverse.movie.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;
import java.util.List;

@Document(collection = "movies")
@Data @Builder @NoArgsConstructor @AllArgsConstructor
public class Movie {

    @Id
    private String id;

    @Indexed(unique = true)
    private String imdbId;

    private String title;
    private String year;
    private String genre;
    private String director;
    private String actors;
    private String plot;
    private String posterUrl;
    private String runtime;
    private String imdbRating;
    private String rated;
    private String language;
    private String country;
    private String awards;
    private String boxOffice;
    private List<Rating> ratings;

    @Builder.Default
    private Instant createdAt = Instant.now();

    @Data @AllArgsConstructor @NoArgsConstructor
    public static class Rating {
        private String source;
        private String value;
    }
}
